src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -322,6 +322,11 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
                     ? PrepareAndEnumerateStatements()
                     : _preparedStatements)
                 {
+                    if (!unprepared)
+                    {
+                        raw.sqlite3_reset(stmt);
```

Comment:
(My goal here is to reset it as soon as we're done with it rather than just before we need to re-use it.)